api/client/login.go

```diff
@@ -74,6 +75,10 @@ func (cli *DockerCli) CmdLogin(args ...string) error {
 			username = authconfig.Username
```

Comment:
Hi, @duglin  Becasue line 86 also use this value. So I do the trimming in line 64.